src/core/AgnLocusStream.c

```diff
@@ -364,9 +365,12 @@ static void locus_stream_extend(AgnLocusStream *stream, AgnLocus *locus)
 
       if(stream->endmode <= 0 && !stream->skip_iiLoci)
       {
-        AgnLocus *iilocus = agn_locus_new(seqid);
         GtRange irange = { prev_range.end + stream->delta + 1,
                            locusrange.start - stream->delta - 1 };
+	//If the two iLoci overlap, do not create an iiLocus:
+        if(irange.start > irange.end) goto noiilocus;
+
```

Comment:
If I'm reading things clearly, this is when handling internal loci. This will prevent the inference of ziLoci, will it not? The details aren't fresh in my mind, but some of the genome organization statistics depended on tracking zero-length iiLoci. Maybe I'm missing something, but it looks like this would affect that tracking.

Reply:
I found this searching in the GitHub repo's history for ziLocus: https://github.com/BrendelGroup/AEGeAn/commit/b2f4d2924c1568a4ee6685c9bba5c9704c48ea2c.

Reply:
The changes to the accounting had the following in mind:
1) If two intervals overlap, then split the overlap evenly for determination of effective length.
2) The sum of effective lengths should add up to the overall sequence lengths in any bounded region (be the entire sequence, the entire genome, or any region bounded by iiLoci).
3) A user should be able to select delta=0 just as well as delta=1, delta=100, or delta=500.

These principles were extensively tested on the Scer genome (which is compact).

Creation of ziLoci occurs elsewhere in the code, I believe, and we need to check whether it is necessary and done correctly.

Reply:
Fair enough.